tulip/abstract/find_controller.py

```diff
@@ -279,10 +279,30 @@ def get_input(
                 ] += mid_weight * np.eye(n)
 
                 r[idx, 0] += -mid_weight * xc
-                u, cost = get_input_helper(
-                    x0, ssys, P1, P3, N, R, r, Q, ord,
-                    closed_loop=closed_loop, solver=solver
-                )
+                try:
+                    u, cost = get_input_helper(
+                        x0, ssys, P1, P3, N, R, r, Q, ord,
+                        closed_loop=closed_loop, solver=solver
+                    )
+                except Exception as ex:
+                    # The end state might consist of several polytopes.
+                    # For some of them there might not be a control action that
+                    # brings the system there. In that case the matrix
+                    # constructed by get_input_helper will be singular and the
+                    # LP solver cannot return a solution.
+                    # This is not a problem unless all polytopes in the end
+                    # region are unreachable, in which case it seems likely that
+                    # there is something wrong with the abstraction routine.
+                    logger.info(repr(ex))
+                    logger.info(
+                        "Failed to find control action from continuous " + \
+                        "state " + str(x0) + " in discrete state " + \
+                        str(start) + " to a target polytope in the " + \
+                        "discrete state " + str(end) + ".\n" + \
+                        "Target polytope:\n" + \
+                        str(P3))
+                    u = 0.0
```

Comment:
From the assignment `low_u = np.zeros([N, m])` [above](https://github.com/tulip-control/tulip-control/pull/242/files#diff-db73a44c4631fa5fc7c04bff0f7d729aR268) the dimension of `u` seems to be `[N, m]`.

`low_cost` is the minimal value of cost over the polytopes that have been iterated over, because initially `low_cost = np.inf` ([above](https://github.com/tulip-control/tulip-control/pull/242/files#diff-db73a44c4631fa5fc7c04bff0f7d729aR267) before the iteration) and `low_cost` changes to only smaller values [below](https://github.com/tulip-control/tulip-control/pull/242/files#diff-db73a44c4631fa5fc7c04bff0f7d729aR308) (if `cost < low_cost`).

On the [next line](https://github.com/tulip-control/tulip-control/pull/242/files#diff-db73a44c4631fa5fc7c04bff0f7d729aR305) `cost = np.inf`, so `cost < low_cost` is `False` [below](https://github.com/tulip-control/tulip-control/pull/242/files#diff-db73a44c4631fa5fc7c04bff0f7d729aR308), and this value of `u` is unused. So `u` could also be left undefined here.